### PR TITLE
Update Prerelease version to Beta and increment patch version.

### PR DESCRIPTION
**Pull request by gkhanna79**
Fixes https://github.com/dotnet/coreclr/issues/5883

@ericstj PTAL.

CC @wtgodbe @dagood @pgavlin

### CODE REVIEW

File: dir.props [hunk 1]
```diff
@@ -151,7 +151,7 @@
 
     <ProjectUrl>https://dot.net</ProjectUrl>
 
-    <!-- PreReleaseSuffix for TFS win8-arm CoreCLR package -->
+    <!-- PreReleaseSuffix for packages published from closed build (e.g. CoreCLR for Arm32, APISet, etc) -->
     <ExternalExpectedPrerelease>rc4-24220-00</ExternalExpectedPrerelease>
```
Comment (ericstj) [inline dir.props]: Leaving this number as-is will create invalid dependencies.  I primed myget with packages that have version beta-24220-00.  Can you use this instead?

Comment (gkhanna79) [inline dir.props]: Done.

**Comment by ericstj** [comment]
LGTM

**Comment by ericstj** [comment]
I see most of the other packages were already addressed with my TFS change.  We might want to delete the versions that are already on myget with the incremented versions and rc4 moniker.

**Comment by gkhanna79** [comment]
I dont have rights to do so. @weshaggard Can you please help clean that up?

**Comment by ericstj** [comment]
I can do that.  LGTM.

File: src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/Microsoft.NETCore.Runtime.CoreCLR.pkgproj [hunk 2]
```diff
@@ -13,16 +13,14 @@
...
     <!-- ApiSets are only applicable for Windows. 
          Despite the unconditioned package dependency it is constrained by runtime IDs within it's own package -->
     <Dependency Include="Microsoft.NETCore.Windows.ApiSets">
-      <Version>1.0.1-rc3-23915</Version>
-    </Dependency>
-    <Dependency Include="Microsoft.NETCore.Jit">
-      <Version>1.0.2$(VersionSuffix)</Version>
+      <Version>1.0.2-$(ExternalExpectedPrerelease)</Version>
     </Dependency>
+    <ProjectReference Include="..\Microsoft.NETCore.Jit\Microsoft.NETCore.Jit.pkgproj" />
```
Comment (gkhanna79) [inline src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/Microsoft.NETCore.Runtime.CoreCLR.pkgproj]: @ericstj This is to line up references similar to what you changes for release branch. Just FYI.

Comment (ericstj) [inline src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/Microsoft.NETCore.Runtime.CoreCLR.pkgproj]: Understood.

**Comment by dagood** [comment]
I fetched this and tried it locally on my (win7) machine and got a `Microsoft.NETCore.ILDAsm` `1.0.3-beta-24220-0` that depends on `Microsoft.NETCore.Runtime.CoreCLR` `1.0.2-beta-24220-0` It looks like the `2` needs to be bumped to `3` there, and in `ILAsm` too.

**Comment by gkhanna79** [comment]
@dagood I didnt realize https://github.com/dotnet/coreclr/issues/5668 was not fixed. I have fixed the issue and confirmed the packages get the correct reference.

@pgavlin PTAL for the update to ILDasm/ILAsm packages to fix https://github.com/dotnet/coreclr/issues/5668.

**Comment by gkhanna79** [comment]
19:55:41 java.lang.ArrayIndexOutOfBoundsException: 0
19:55:41 	at hudson.plugins.git.GitChangeSet.findOrCreateUser(GitChangeSet.java:377)
19:55:41 	at hudson.plugins.git.GitChangeSet.getAuthor(GitChangeSet.java:435)
19:55:41 	at hudson.model.AbstractBuild.getCulprits(AbstractBuild.java:350)
19:55:41 	at hudson.model.AbstractBuild.getCulprits(AbstractBuild.java:346)
19:55:41 	at hudson.model.AbstractBuild$AbstractBuildExecution.post(AbstractBuild.java:672)
19:55:41 	at hudson.model.Run.execute(Run.java:1763)
19:55:41 	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
19:55:41 	at hudson.model.ResourceController.execute(ResourceController.java:98)
19:55:41 	at hudson.model.Executor.run(Executor.java:410)
19:55:41 [BFA] Scanning build for known causes...
19:55:42 .........................................................................[BFA] No failure causes found
19:57:04 [BFA] Done. 82s

@mmitche Is this an infra issue in CentOS build?

**Comment by gkhanna79** [comment]
@dotnet-bot test CentOS7.1 x64 Debug Build and Test  please

**Comment by gkhanna79** [comment]
@dotnet-bot test Windows_NT x86 legacy_backend Checked Build and Test please

**Comment by gkhanna79** [comment]
@dotnet-bot test Windows_NT x86 ryujit Checked Build and Test please

**Comment by mmitche** [comment]
@gkhanna79 Infra issue there.  Well, git issue sorta.  Did you happen to push a commit from a machine with no email configured (or an invalid email address)?

It appears that ocassionally we see email addresses on commits sent as just '@'.  This is parsed to obtain a user name in cases where an official GH provided user name wasn't available.

I made and submitted a fix to whatever plugin it is that has this issue and it got accepted, but we haven't upgraded yet (this hits very rarely)

**Comment by mmitche** [comment]
@gkhanna79 What other issues did you see in the other failed legs?

**Comment by gkhanna79** [comment]
They were same as the one I posted.

**Comment by mmitche** [comment]
@gkhanna79 Yeah there is a misconfigured email address somewhere in the massive list of commits that it thought it was testing

Better question is how did it think there were > 1000 commits the first time around.  Did you have an odd rebase here?

**Comment by gkhanna79** [comment]
> Better question is how did it think there were > 1000 commits the first time around. Did you have an odd rebase here?

This is very strange - if the rebase was off, wouldnt it show in the "Files changed" section above?

**Comment by gkhanna79** [comment]
@mmitche Does it still show the >1000 commits in the latest version of the branch I have pushed?

**Comment by mmitche** [comment]
@gkhanna79 No.  It may have been because the previous build of this job was a very different commit (it often measures the deltas in commits between jobs).  However, it did show up when there was only one commit:

http://dotnet-ci.cloudapp.net/job/dotnet_coreclr/job/master/job/debug_centos7.1_prtest/2546/console

The failure is just a string split operation on '@' over an email address, indexed at [0].  So if you have an invalid email address this can happen.  Such an email address could either come from git, or perhaps from bad data from GH.

**Comment by mmitche** [comment]
@gkhanna79 It may have been the combination of the two.  Here is the culprit (from git log output):

PS C:\enlistments\coreclr> git log | findstr '<@>'
Author: Maoni0 <@>
Author: Maoni0 <@>
Author: Maoni0 <@>

**Comment by gkhanna79** [comment]
Thanks for looking into this @mmitche.

@ericstj I will merge this once @pgavlin takes a look.

**Comment by pgavlin** [comment]
LGTM.

**Pull request merged**